packages/ava/src/insight/chart/generator/insights.ts

```diff
@@ -51,6 +51,13 @@ export function generateInsightChartSpec(insight: InsightInfo<PatternInfo>): G2S
     correlation: correlationStrategy,
   };
 
+  // majority insight pattern visualizes as 'pie', should not use y nice (G2 handle it as rescale y, the pie chart will be less than 100%)
+  const isThetaCoordinate = insight.patterns.map((pattern) => pattern.type).includes('majority');
+  const viewConfig = isThetaCoordinate
+    ? { scale: { y: { nice: false } } }
+    : {
+        scale: { y: { nice: true } },
+      };
   const marks = insightType2Strategy[insightType]?.(insight);
-  return viewSpecStrategy(marks);
+  return viewSpecStrategy(marks, viewConfig);
```

Comment:
moving isThetaCoordinate related codes into viewSpecStrategy is more suitable

Reply:
`viewSpecStrategy` receive marks as its params, it is kind of hard to infer whether the spec has theta coordinate mark, because each mark may also have children. So just use the pattern type to infer the `nice` property

Reply:
> `viewSpecStrategy` receive marks as its params, it is kind of hard to infer whether the spec has theta coordinate mark, because each mark may also have children. So just use the pattern type to infer the `nice` property

After discussing, move the logic to viewSpecStrategy